server/channels/app/server.go

```diff
@@ -1871,3 +1868,125 @@ func (s *Server) Log() *mlog.Logger {
 func (s *Server) NotificationsLog() *mlog.Logger {
 	return s.platform.NotificationsLogger()
 }
+
+func (a *App) initElasticsearchChannelIndexCheck() {
+	a.elasticsearchChannelIndexCheckWithRetry()
+
+	a.AddConfigListener(func(oldConfig, newConfig *model.Config) {
+		if a.SearchEngine().ElasticsearchEngine == nil {
+			return
+		}
+
+		oldESConfig := oldConfig.ElasticsearchSettings
+		newESConfig := newConfig.ElasticsearchSettings
+
+		// if indexing is turned on, check.
+		if !*oldESConfig.EnableIndexing && *newESConfig.EnableIndexing {
+			a.elasticsearchChannelIndexCheckWithRetry()
+		} else if *newESConfig.EnableIndexing && (*oldESConfig.Password != *newESConfig.Password || *oldESConfig.Username != *newESConfig.Username || *oldESConfig.ConnectionURL != *newESConfig.ConnectionURL || *oldESConfig.Sniff != *newESConfig.Sniff) {
+			// ES client reconnects if credentials or address changes
+			a.elasticsearchChannelIndexCheckWithRetry()
+		}
+	})
+
+	a.AddLicenseListener(func(oldLicense, newLicense *model.License) {
+		if a.SearchEngine() == nil {
+			return
+		}
+
+		// if a license was added, and it has ES enabled-
+		if oldLicense == nil && newLicense != nil {
+			if a.SearchEngine().ElasticsearchEngine != nil {
+				a.elasticsearchChannelIndexCheckWithRetry()
+			}
+		}
+	})
+}
+
+func (a *App) elasticsearchChannelIndexCheckWithRetry() {
+	go func() {
```

Comment:
Actually it always needs to be called async, other license application or config update will take a time., because those listeners are called synchronusly. I was calling this sync initially but had to make it async due to those issues. Even locally,, it was causing noticeable delays in license application and config updation.

Reply:
ok